### PR TITLE
Capability check for `s3_plain`

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

The capability check was removed for some reason for `s3_plain`.

https://github.com/ClickHouse/ClickHouse/pull/58357

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit e112288c0b47d793d5c1a732655c4ae240b526ca with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59145/e112288c0b47d793d5c1a732655c4ae240b526ca/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59145/e112288c0b47d793d5c1a732655c4ae240b526ca/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Disks/ObjectStorages/ObjectStorageFactory.cpp [hunk 1]
```diff
@@ -159,8 +165,14 @@ void registerS3PlainObjectStorage(ObjectStorageFactory & factory)
         auto client = getClient(config, config_prefix, context, *settings);
         auto key_generator = getKeyGenerator(disk_type, uri, config, config_prefix);
 
-        return std::make_shared<S3PlainObjectStorage>(
+        auto object_storage = std::make_shared<S3ObjectStorage>(
```
Comment (kssenii) [inline src/Disks/ObjectStorages/ObjectStorageFactory.cpp:168]: why `S3PlainObjectStorage` -> `S3ObjectStorage`?

Comment (antonio2368) [inline src/Disks/ObjectStorages/ObjectStorageFactory.cpp:168]:
```suggestion
        auto object_storage = std::make_shared<S3PlainObjectStorage>(
```

Comment (antonio2368) [reply src/Disks/ObjectStorages/ObjectStorageFactory.cpp:168]: bad c/p 😓

**Review by kssenii — APPROVED** [review]

**Pull request merged**